### ISSUE TITLE
Watch: Wait for first compilation to succeed

### ISSUE THREAD

**Issue by timm-gs** [whoeverest/wsrun#1]
Hi,

I found this through your link in https://github.com/Microsoft/TypeScript/issues/3469#issuecomment-348162840

what would be super useful in TypeScript `watch` scenarios, would be to wait for `watch` to complete the first compile before starting on the next (possibly parent) file.

As `tsc` does not clean up the `outDir`, all our `watch` commands are prefixes with a command to delete the output directory, such that we start from a clean base and would get compile errors immediately if some packages imports a file which source (TS) no longer exists.

And if a watch exists on a linked project which output dir is removed, `tsc` gets irrevocably broken.

Therefore it would be a great help if `watch`s could be started in sequence.

Not sure how TypeScript specific/helpful you would like to make this project, just an idea for now.

**Comment by timm-gs** [comment]
It just occurred to me, that we might be able to work around removing the output dir while some `tsc` instances are already watching it, by doing it before starting the individual watch compilations.

I will try your tool with our setup later to investigate this. But unless it does something special to wait, I would expect the some initial `watch` invocation to fail.

But they might be eventually converge into a stable whole?

**Comment by spion** [comment]
I'd imagine I'd have a "clean-and-watch" workspace level command that does:

```json
{
  "clean-and-watch": "wsrun clean && wsrun build --stages && wsrun watch-without-clean"
}
```

then you can `yarn run clean-and-watch`

Slower than a normal watch, but should work. 

In terms of how to do this without slowing down, I suppose it could have pluggable "criteria" for when a process is actually done - something like `--doneCriteria "Compilation complete.+watching"`

**Comment by timm-gs** [comment]
@spion Thanks, that's actually a clever order to run things, works great :)

Also works nicely with `lerna` (`lerna run clean && lerna run build && lerna run --parallel watch-only`), though I am looking forward to try the `wsrun` `[<package>]` option in order to not build packages that are not needed for the task at hand.

So my immediate problem is solved, but since now at least 3 people seem interested in such a `doneCriteria` flag I will keep this open for now.

**Comment by spion** [comment]
@timm-gs --done-criteria has been added in the latest version.
